python/pyarrow/includes/libarrow.pxd

```diff
@@ -794,6 +794,10 @@ cdef extern from "arrow/api.h" namespace "arrow" nogil:
         shared_ptr[CRecordBatch] Slice(int64_t offset)
         shared_ptr[CRecordBatch] Slice(int64_t offset, int64_t length)
 
+    cdef cppclass CRecordBatchWithMetadata" arrow::RecordBatchWithMetadata":
+        shared_ptr[CRecordBatch] batch
+        const shared_ptr[const CKeyValueMetadata] custom_metadata
```

Comment:
In C++, there is no `const` qualifier for this field. However, Cython reports some const assignment failure due to this, I read the code, and found several similar TODO items (https://github.com/apache/arrow/blob/2a2c0873b3ae62d2d08225cc88e53ae004864a7f/python/pyarrow/includes/libarrow.pxd#L415, https://github.com/apache/arrow/blob/2a2c0873b3ae62d2d08225cc88e53ae004864a7f/python/pyarrow/includes/libarrow.pxd#L462) about this part. 

I am not familiar with Cython at all, and have to add the `const` qualifier to make it work at first. Please advise how this could be revised. Thanks.

Reply:
I think it's ok to add the `const`. You can add a comment explaining this trick.

Reply:
Thanks. Comment is added to explain this.